File: javascript/digitSum.js
values = [];

function digitSum(string){
	var sum = eval(string.split('').join("+"));
	values.push(sum.toString());
	if(sum.toString().length != 1){
		digitSum(sum.toString())
	}
	return values[values.length - 1];
}
// 
debug(digitSum("2468"))

File: ruby/digit_sum.rb
@values = []

def digit_sum(string)
  sum = string.split("").map!{|num| num.to_i}.inject(:+)
    @values << sum.to_s
    if sum.to_s.length != 1
       digit_sum(sum.to_s)
    end
    @values.last
end

print digit_sum("999999787")
